ftdc/ftdc.go

```diff
@@ -232,7 +244,9 @@ func (ftdc *FTDC) statsWriter() {
 }
 
 // StopAndJoin stops the background worker started by `Start`. It is only legal to call this after
-// `Start` returns.
+// `Start` returns. It's normal for tests to _not_ call `StopAndJoin`. Tests that have spun up the
+// `statsWriter` by hand, without the `statsReader` can `close(ftdc.datumCh)` followed by
+// `<-ftdc.outputWorkerDone` to stop+wait for the `statsWriter`.
 func (ftdc *FTDC) StopAndJoin(ctx context.Context) {
```

Comment:
Shouldn't `StopAndJoin` close the current output file?

Reply:
No, `statsWriter` might still be running and thus it owns the file object and is responsible for closing.